### PR TITLE
Internal emscripten build cleanup

### PR DESCRIPTION
**Pull request by ekpyron**
Parts of https://github.com/ethereum/solidity/pull/7832

### CODE REVIEW

File: cmake/EthCompilerSettings.cmake [hunk 1]
```diff
@@ -110,9 +110,12 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
 			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s STRICT=1")
 			# Export the Emscripten-generated auxiliary methods which are needed by solc-js.
 			# Which methods of libsolc itself are exported is specified in libsolc/CMakeLists.txt.
-			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s EXTRA_EXPORTED_RUNTIME_METHODS=['cwrap','addFunction','removeFunction','Pointer_stringify','lengthBytesUTF8','_malloc','stringToUTF8','setValue']")
+			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s EXTRA_EXPORTED_RUNTIME_METHODS=['cwrap','addFunction','removeFunction','UTF8ToString','lengthBytesUTF8','_malloc','stringToUTF8','setValue']")
```
Comment (ekpyron) [inline cmake/EthCompilerSettings.cmake]: @axic Do we want this here or do we want to move this to https://github.com/ethereum/solidity/pull/7832 ?

Comment (axic) [reply cmake/EthCompilerSettings.cmake]: I think this part of the emscripten update, so #7832.

**Review by axic — DISMISSED** [review]

**Comment by ekpyron** [comment]
The external test failure might be related to https://github.com/OpenZeppelin/openzeppelin-contracts/commit/5f92adc2e76fe92d7ab952710ff3fb6d76066a35

**Comment by ekpyron** [comment]
I think the tests will depend on https://github.com/ethereum/solidity/pull/7845

**Pull request merged**